perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2495,64 +2396,43 @@ private void checkDuplicateRegistrationAttempt(PerunSession sess, AppType appTyp
 				} catch (MemberNotExistsException ex) {
 					// user is not member of vo
 					if (group != null) {
-						// not member of VO - check for unprocessed applications to Group
-						checkDuplicateRegistration("Initial application for Group: "+group.getName()+" already exists.",
-								AppType.INITIAL, vo.getId(), group.getId(), user.getId(), actor, extSourceName);
+						checkDupplicateGroupApplications(sess, vo, group, AppType.INITIAL);
+						//throw new InternalErrorException("You must be member of vo: "+vo.getName()+" to apply for membership in group: "+group.getName());
 					} else {
-						// not member of VO - check for unprocessed applications
-						checkDuplicateRegistration("Initial application for Vo: "+vo.getName()+" already exists.",
-								AppType.INITIAL, vo.getId(), 0, user.getId(), actor, extSourceName);
+						checkDupplicateVoApplications(sess, vo, AppType.INITIAL);
+						// pass not member and have only approved or rejected apps
 					}
 				}
 			} else {
-
 				// user is not known
 				if (group != null) {
-					// group application
-					checkDuplicateRegistration("Initial application for Group: "+group.getName()+" already exists.",
```

Comment:
Private method ``checkDuplicateRegistration()`` is no longer used and its implementation should be removed too.

Reply:
removed